From 9929494d3d1c4f4a99b1a1633ae0874f6b0ae445 Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Sun, 13 Jan 2013 11:26:11 +0100
Subject: [PATCH] Allow tests to run with <python2.7

runtests.py trick does not seem to work that well.
At the very least it misses assertLessEqual so just use unittest2 when
necessary and be done with hacks.
---
 tests/runtests.py             | 30 +++---------------------------
 tests/test_atoms.py           |  3 +++
 tests/test_everything.py      |  4 ++++
 tests/test_gi.py              |  6 +++++-
 tests/test_iochannel.py       |  4 ++++
 tests/test_overrides.py       |  3 +++
 tests/test_overrides_gdk.py   |  3 +++
 tests/test_overrides_gtk.py   |  4 ++++
 tests/test_overrides_pango.py |  3 +++
 tests/test_pygtkcompat.py     |  3 +++
 10 files changed, 35 insertions(+), 28 deletions(-)

diff --git a/tests/runtests.py b/tests/runtests.py
index c518fcc..f0a6750 100755
--- a/tests/runtests.py
+++ b/tests/runtests.py
@@ -5,34 +5,10 @@ import os
 import glob
 import sys
 
-import unittest
-
-# provide missing unittest decorators and API for python 2.6; these decorators
-# do not actually work, just avoid the syntax failure
 if sys.version_info[:2] == (2, 6):
-    def skipUnless(condition, reason):
-        if not condition:
-            sys.stderr.write('[expected failure] ')
-        return lambda obj: obj
-
-    unittest.skipUnless = skipUnless
-    unittest.expectedFailure = lambda obj: obj
-
-    def assertGreater(self, a, b, msg=None):
-        if not a > b:
-            self.fail('%s not greater than %s' % (repr(a), repr(b)))
-
-    def assertGreaterEqual(self, a, b, msg=None):
-        if not a >= b:
-            self.fail('%s not greater than or equal to %s' % (repr(a), repr(b)))
-
-    def assertIsInstance(self, obj, cls, msg=None):
-        if not isinstance(obj, cls):
-            self.fail('%s is not an instance of %r' % (repr(obj), cls))
-
-    unittest.TestCase.assertGreaterEqual = assertGreaterEqual
-    unittest.TestCase.assertGreater = assertGreater
-    unittest.TestCase.assertIsInstance = assertIsInstance
+    import unittest2 as unittest
+else:
+    import unittest
 
 if '--help' in sys.argv:
     print("Usage: ./runtests.py <testfiles>")
diff --git a/tests/test_atoms.py b/tests/test_atoms.py
index a59d15a..ff3d9fb 100644
--- a/tests/test_atoms.py
+++ b/tests/test_atoms.py
@@ -1,5 +1,8 @@
 import unittest
 
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 try:
     from gi.repository import Atk, Gdk, Gtk
     (Atk, Gdk)  # pyflakes
diff --git a/tests/test_everything.py b/tests/test_everything.py
index 600edbf..744178f 100644
--- a/tests/test_everything.py
+++ b/tests/test_everything.py
@@ -3,6 +3,10 @@
 # vim: tabstop=4 shiftwidth=4 expandtab
 
 import unittest
+
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 import traceback
 import warnings
 import gc
diff --git a/tests/test_gi.py b/tests/test_gi.py
index 315867a..d74a095 100644
--- a/tests/test_gi.py
+++ b/tests/test_gi.py
@@ -4,7 +4,11 @@
 
 import sys
 
-import unittest
+if sys.version_info[:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+
 import tempfile
 import shutil
 import os
diff --git a/tests/test_iochannel.py b/tests/test_iochannel.py
index 8c035e9..a28e7ff 100644
--- a/tests/test_iochannel.py
+++ b/tests/test_iochannel.py
@@ -2,6 +2,10 @@
 # encoding: UTF-8
 
 import unittest
+
+if not hasattr(unittest.TestCase, 'assertLessEqual'):
+    import unittest2 as unittest
+
 import tempfile
 import os.path
 import fcntl
diff --git a/tests/test_overrides.py b/tests/test_overrides.py
index dd2aa6a..c72e5b6 100644
--- a/tests/test_overrides.py
+++ b/tests/test_overrides.py
@@ -3,6 +3,9 @@
 
 import unittest
 
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 import gi.overrides
 try:
     from gi.repository import Regress
diff --git a/tests/test_overrides_gdk.py b/tests/test_overrides_gdk.py
index 070acb8..16d976f 100644
--- a/tests/test_overrides_gdk.py
+++ b/tests/test_overrides_gdk.py
@@ -3,6 +3,9 @@
 
 import unittest
 
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 import gi.overrides
 
 try:
diff --git a/tests/test_overrides_gtk.py b/tests/test_overrides_gtk.py
index ee41457..c1dbc8f 100644
--- a/tests/test_overrides_gtk.py
+++ b/tests/test_overrides_gtk.py
@@ -2,6 +2,10 @@
 # vim: tabstop=4 shiftwidth=4 expandtab
 
 import unittest
+
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 import ctypes
 import sys
 
diff --git a/tests/test_overrides_pango.py b/tests/test_overrides_pango.py
index 51f4f6a..7468fbb 100644
--- a/tests/test_overrides_pango.py
+++ b/tests/test_overrides_pango.py
@@ -3,6 +3,9 @@
 
 import unittest
 
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 try:
     from gi.repository import Pango
     Pango
diff --git a/tests/test_pygtkcompat.py b/tests/test_pygtkcompat.py
index 5fabf2a..b5ddf89 100644
--- a/tests/test_pygtkcompat.py
+++ b/tests/test_pygtkcompat.py
@@ -3,6 +3,9 @@
 
 import unittest
 
+if not hasattr(unittest, 'skipUnless'):
+    import unittest2 as unittest
+
 from gi.repository import GLib
 
 try:
-- 
1.8.1

